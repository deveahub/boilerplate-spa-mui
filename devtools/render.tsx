import { render as tRender } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import { InitialState, setInitialState } from '../src/state';
import { ThemeProvider } from '../src/theme';

type TRenderParams = Parameters<typeof tRender>;

type RenderProps = {
  initialState?: InitialState
}

const render = (
  node: TRenderParams[0],
  props?: TRenderParams[1],
  renderProps?: RenderProps,
) => {
  const { initialState = {} } = renderProps || {};
  const enhancedNode = (
    <RecoilRoot initializeState={setInitialState(initialState)}>
      <ThemeProvider>
        {node}
      </ThemeProvider>
    </RecoilRoot>
  );

  return tRender(enhancedNode, props);
};

export default render;
