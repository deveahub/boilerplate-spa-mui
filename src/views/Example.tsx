import { Typography } from '@mui/material';
import { useTranslation } from 'react-i18next';

const Example = () => {
  const { t } = useTranslation();
  return (
    <Typography>
      Example
      {t('common:test')}
    </Typography>
  );
};

export default Example;
