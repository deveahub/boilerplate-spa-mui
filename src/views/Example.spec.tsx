import { render } from '../../devtools';
import Example from './Example';

describe('Example', () => {
  it('render', () => {
    const { asFragment } = render(<Example />);
    expect(asFragment()).toMatchSnapshot();
  });
});
