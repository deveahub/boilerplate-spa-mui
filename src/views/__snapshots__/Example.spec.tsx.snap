// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example render 1`] = `
<DocumentFragment>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Examplecommon:test
  </p>
</DocumentFragment>
`;
